src/vstart.sh

```diff
@@ -1041,6 +1047,90 @@ EOF
 
 }
 
+start_ganesha() {
+    GANESHA_PORT=$(($CEPH_PORT + 4000))
+    local ganesha=0
+    for name in a b c d e f g h i j k l m n o p
+    do
+        [ $ganesha -eq $GANESHA_DAEMON_NUM ] && break
+
+        port=$(($GANESHA_PORT + ganesha))
+        ganesha=$(($ganesha + 1))
+        ganesha_dir="$CEPH_DEV_DIR/ganesha.$name"
+
+        prun rm -rf $ganesha_dir
+        prun mkdir -p $ganesha_dir
+
+        echo "NFS_CORE_PARAM {
+        Enable_NLM = false;
+        Enable_RQUOTA = false;
+        Protocols = 4;
+        NFS_Port = $port;
+}
+
+CACHEINODE {
+        Dir_Chunk = 0;
+        NParts = 1;
+        Cache_Size = 1;
+}
+
+NFSv4 {
+        RecoveryBackend = 'rados_cluster';
+        Minor_Versions = 1, 2;
+}
+
+EXPORT {
+	Export_Id = 100;
+	Transports = TCP;
+	Path = /;
+	Pseudo = /ceph/;
+	Protocols = 4;
+	Access_Type = RW;
+	Attr_Expiration_Time = 0;
+	FSAL {
+	    Name = CEPH;
+	}
+}
+
+CEPH {
+	Ceph_Conf = $conf_fn;
+}
+
+RADOS_KV {
+	Ceph_Conf = $conf_fn;
+	pool = 'cephfs.a.meta';
+	namespace = 'ganesha';
+	UserId = 'admin';
+	nodeid = $name;
+}" > "$ganesha_dir/ganesha.conf"
+
+
+	wconf <<EOF
+[ganesha.$name]
+        host = $HOSTNAME
+        ip = $IP
+        port = $port
+        ganesha data = $ganesha_dir
+        pid file = $ganesha_dir/ganesha.pid
+EOF
+        prun $CEPH_BIN/rados lspools
+
+        prun ganesha-rados-grace -p cephfs.a.meta -n ganesha add $name
+        prun ganesha-rados-grace -p cephfs.a.meta -n ganesha
+
+        if !(firewall-cmd --list-services | grep nfs) ; then
+            prun firewall-cmd --add-service=nfs
+            prun firewall-cmd --reload
+        fi
```

Comment:
Given that you aren't running ganesha on standard NFS port, do you need to monkey with the firewall?

Reply:
I did not do any modification to the firewall.

Reply:
*blink*

firewall-cmd does exactly that. You're adding the NFS "service" to it here and reloading it. I don't think that's necessary since you're running this on a random port.

Reply:
Sorry, I meant apart from firewall-cmd.